### PR TITLE
Flav/poke string ids

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR introduces a search command palette in Poke, allowing us to search using string IDs. Since we now use [squids](https://github.com/sqids/sqids-javascript) to generate string IDs, we need an effective way to search for them, as they are no longer stored in our database.

This update provides a palette in Poke for searching with a string ID, currently supporting only data sources and data source views. This is a temporary solution to address our immediate needs. In the future, we'll refine the system by integrating everything behind this endpoint and supporting more sIDs.

 It's not the perfect solution, as we don't have the workspace's sID, which necessitates making `toPokeJSON` asynchronous...

PokeSquids

### CODE REVIEW

File: front/components/poke/PokeNavbar.tsx [hunk 1]
```diff
@@ -1,8 +1,17 @@
 import { Logo } from "@dust-tt/sparkle";
 import { isDevelopment } from "@dust-tt/types";
 import Link from "next/link";
+import { useEffect, useState } from "react";
 
+import { PokeButton } from "@app/components/poke/shadcn/ui/button";
```
Comment (Fraggle) [inline front/components/poke/PokeNavbar.tsx:6]: "shadcn" ?

File: front/components/poke/PokeNavbar.tsx [hunk 2]
```diff
@@ -24,7 +33,71 @@ export const PokeNavbar: React.FC = () => (
...
+
+  const { isError, isLoading, results } = usePokeSearch({
+    // Disable search until the user has typed at least 2 characters.
+    disabled: searchTerm.length < 2,
+    search: searchTerm,
+  });
+
+  useEffect(() => {
+    const down = (e: KeyboardEvent) => {
+      if (e.key === "k" && (e.metaKey || e.ctrlKey)) {
```
Comment (Fraggle) [inline front/components/poke/PokeNavbar.tsx:54]: what about "/" as well ?

File: front/components/poke/PokeNavbar.tsx [hunk 3]
```diff
@@ -24,7 +33,71 @@ export const PokeNavbar: React.FC = () => (
...
+        shouldFilter={false}
+      >
+        <PokeCommandInput
+          placeholder="Type a command or search..."
+          onValueChange={(value) => setSearchTerm(value)}
+          className="border-none focus:outline-none focus:ring-0"
+        />
+        <PokeCommandList>
+          {isLoading && <div className="p-4 text-sm">Searching...</div>}
+          {!isError && !isLoading && results.length === 0 && (
```
Comment (Fraggle) [inline front/components/poke/PokeNavbar.tsx:81]: nit && searchTerms && searchTerms.length >= 2

and a special / no message when nothing has been searched

**Review by JulesBelveze — APPROVED** [review]
LGTM

As a general note, I'm not sure to understand why we have components on steroids for poke and not for the rest? 🤔 
Also weird to maintain a parallel design system for poke..

**Comment by tdraier** [comment]
🔥 

I'm not very fond of having toPokeJSON() and poke types directly inside the resources, we could keep all poke logic outside of the general codebase. but otherwise LGTM

**Review by Fraggle — APPROVED** [review]
🔥 EXCELLENT 🔥 

Poke stuff / types are spawning a little bit everywhere, do we really it ?

What about just sending whatever we need from the poke search endpoint, everything else should fall into place as we use the return type of the endpoint in the frontend anyway.

Can you also add all the other sids ? (vault, workspace, connector...)

File: front/components/poke/PokeNavbar.tsx [hunk 4]
```diff
@@ -24,7 +33,71 @@ export const PokeNavbar: React.FC = () => (
...
+        className="bg-structure-50 sm:max-w-[600px]"
+        shouldFilter={false}
+      >
+        <PokeCommandInput
+          placeholder="Type a command or search..."
+          onValueChange={(value) => setSearchTerm(value)}
+          className="border-none focus:outline-none focus:ring-0"
+        />
+        <PokeCommandList>
+          {isLoading && <div className="p-4 text-sm">Searching...</div>}
```
Comment (Fraggle) [inline front/components/poke/PokeNavbar.tsx:80]: nit: we have a nice spinner anim :)

**Comment by flvndvd** [comment]
Seen IRL, went ahead and consolidated all types in a dedicated poke file for now. Will revisit the `toPokeJSON` in the coming days when adding more resource behind it.

**Pull request merged**